### ISSUE TITLE
[semi] Semicolons aren't checked for type aliases

### ISSUE THREAD

**Issue by slikts** [typescript-eslint/typescript-eslint#366]
**Repro**

```TS
type Foo = {}
type Bar = {};
```

**Expected Result**
One of the lines should be wrong, but they're both valid, regardless of ESLint config.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.1` |
| `@typescript-eslint/parser`        | `1.4.1` |

**Comment by bradzacher** [comment]
[the base rule has a hardcoded list of nodes it checks.](https://github.com/eslint/eslint/blob/master/lib/rules/semi.js#L305-L325)

We'll have to extend the base rule if we want it to support our `TSTypeAliasDeclaration` node.

A lot of users use prettier, so I guess this has gone unnoticed for so long because of that.

**Comment by JamesHenry** [comment]
Fixed by #461

**Comment by bopfer** [comment]
Any estimate on when the next release will happen?  This is something we would like to get resolved in our codebase.

**Comment by JamesHenry** [comment]
@bopfer As per our README, we release on every single commit to master, so you never need to wait for a fix or feature: https://github.com/typescript-eslint/typescript-eslint#package-versions

**Comment by JamesHenry** [comment]
We update the changelog and update the latest tag on an ad hoc basis ever couple of weeks/when necessary, but it should hopefully never be a blocker

**Comment by bopfer** [comment]
Hmm, I am confused.  1.7.0 is the latest version on NPM: https://www.npmjs.com/package/@typescript-eslint/eslint-plugin and it does not have this fix.  Maybe I am missing something.  I see lots of alpha versions since then, but I am not using those.

**Comment by JamesHenry** [comment]
Exactly - the `latest` tag on npm gets updated every now and again, but the fixes are always available in the `canary` tag, which are the other versions you are referencing. We release on every single commit to master you _always_ have an installable package available to you.

You can install any one of those from the point at which this fix was merged and the fix will be in there.

That same code will be in the `latest` tag on npm at some point in the future, but it will not be any different.

**Comment by bopfer** [comment]
Ok.  I get it now.  I just need a little more patience :smile:  Thanks for the clarification.
